pycti/utils/opencti_stix2_splitter.py

```diff
@@ -96,16 +93,91 @@ def by_dep_size(elem):
             return elem["nb_deps"]
 
         self.elements.sort(key=by_dep_size)
-        for entity in self.elements:
+
+        # Algorithm to re-bundle elements
+        # Bundle created group the relationships by dependencies
+        # This grouping will improve relations sharding
+        if relations_grouping:
+            cache_ids = {}
+            id_alias = {}
+            ids_equivalence = {}
+            working_elements = {}
+            grouped_elements = {}
+            for elem in self.elements:
+                if cache_ids.get(elem["id"]):
+                    # Prevent sending duplicate elements for ingest
+                    continue
+                nb_deps = elem["nb_deps"]
+                if elem["type"] == "relationship":
+                    source_ref = (
+                        ids_equivalence.get(elem["source_ref"], elem["source_ref"])
+                        + "-"
+                        + str(nb_deps)
+                    )
+                    target_ref = (
+                        ids_equivalence.get(elem["target_ref"], elem["target_ref"])
+                        + "-"
+                        + str(nb_deps)
+                    )
+                    target_source = id_alias.get(source_ref)
+                    target_target = id_alias.get(target_ref)
+
+                    target = "generated-" + str(uuid.uuid4())
```

Comment:
what is "generated-" + str(uuid.uuid4()) ? an ID ? it's the first time I see "generated" as an ID prefix, do we handle it somewhere else in the code ?

Reply:
Its a generic id only used to group information. This id will not be propagated as only values will remains after the processing (grouped_elements.values())